packages/compile-solidity/index.js

```diff
@@ -86,27 +90,27 @@ const Compile = {
 
     options = Config.default().merge(options);
 
+    debug("invoking profiler");
     const { allSources, compilationTargets } = await Profiler.requiredSources(
       options.with({
         paths,
         base_path: options.contracts_directory,
         resolver: options.resolver
       })
     );
+    debug("allSources: %O", allSources);
+    debug("compilationTargets: %O", compilationTargets);
 
     // we can exit if there are no Solidity files to compile since
     // it indicates that we only have Vyper-related JSON
-    const solidityTargets = compilationTargets.filter(fileName => {
-      return fileName.endsWith(".sol");
-    });
-    if (compilationTargets.length !== 0 && solidityTargets.length === 0) {
+    const solidityTargets = compilationTargets.filter(fileName =>
+      fileName.endsWith(".sol")
+    );
+    if (solidityTargets.length === 0) {
       return { compilations: [] };
     }
-    const hasTargets = compilationTargets.length;
 
-    hasTargets
-      ? Compile.display(compilationTargets, options)
-      : Compile.display(allSources, options);
+    Compile.display(compilationTargets, options);
```

Comment:
I made it so that the profiler will no longer return `[]` for the targets to indicate compile everything; instead it will just return everything as the target.  So, no more need for a case split here.

...ah, crap, I just realized I forgot to mention that at all in the PR writeup.  Guess I'll go back and edit that in...

Reply:
I went back and re-read the whole writeup to see if I could find the explanation :)

Reply:
Ha, I just stuck it at the end.  This PR, and its writeup, was a mess, and I had a hard time trying to explain it coherently.  So no way was I going to try to edit this in somewhere good in the body!